### PR TITLE
MCAS-31: enable custom secondary indices

### PR DESCRIPTION
**Pull request by dwaddington**
Added bool iterate_key_space_on_load which can be used by persistent indexes to avoid rebuild.  However, this is temporary, since we should have a way to check the key-spaces are equivalent.  Iterating the whole space sort of defeats the object of using persistence.  
Anyway, these changes are sufficient for some initial tests.

### CODE REVIEW

**Comment by dwaddington** [comment]
Perhaps because on a rebuild we would be only doing a read of the key space, iteration is ok, since time(reading+checking) < time(writing+reconstructing).

File: src/components/api/kvindex_itf.h [hunk 1]
```diff
@@ -118,23 +133,25 @@ class IKVIndex_factory : public component::IBase {
...
-                           const std::string& param,
-                           const std::string& param2)
+  /** 
+   * create_configured is used when the secondary index is defined in configuration file 
+   * 
+   * @param primary_index 
+   * 
+   * @return 
+   */
+  virtual IKVIndex* create_configured(IKVStore * primary_index)
```
Comment (moshik1) [inline src/components/api/kvindex_itf.h:150]: why it is primary)index and not secondary index?

**Review by moshik1 — COMMENTED** [review]
in index_itf.h line 150: why it is a primary_index and not a secondary index?

File: src/components/api/kvindex_itf.h (hunk 1 above)
Comment (dwaddington) [reply src/components/api/kvindex_itf.h:150]: Hstore, Hstore-CC and Mapstore are primary indexes, i.e. the hash table.

Comment (moshik1) [reply src/components/api/kvindex_itf.h:150]: OK, understood

**Review by moshik1 — APPROVED** [review]

**Pull request merged**